### ISSUE TITLE
Random games asking for nonexistent updates.

### ISSUE THREAD

**Issue by Zetchzie** [roxas75/rxTools#372]
So im using the latest git and im on 10.5 o3ds and at random some games will ask to update but there is no update for the game. if you let it try to download said update it trys to download then stops with an error to check connection. if you go to eshop no such update exists.

**Comment by ApacheThunder** [comment]
Assuming you have installed them via CIAs, It's because the CIA was packed with the incorrect version string. Perhaps this wasn't as big an issue in older firmwares. It's an issue more with the CIA then with rxTools. For example, I repacked Hatsune Miku Project Mirai DX with the CIA version set to 0.1.0. Many repacks you get from the usual places may have had this set to 0.0.0.

Find out what the normal version it should be at and repack the CIA with that version and reinstall the games in question. This is not an issue rxTools can fix as trying to fix this will only break other games that may infact need an update. :P

**Comment by urherenow** [comment]
Which games, and are you sure it's asking for a game update and not a SYSTEM update?

**Comment by ApacheThunder** [comment]
I've had a game do this. I'm pretty sure he means the games. I had Hatsune Miku Project Mirai do this. Repacked CIA with correct version set and it doesn't happen with that game anymore. (I haven't noticed any other games asking to update yet).

The reason it's failing because the games in question don't have an update, so it tries to download the full game. Assuming they were installed as CIAs and are not legit, you won't have a valid ticket, so eShop kicks you out after reaching 99%. :P

**Comment by Shadow322** [comment]
One game I noticed was angry video game nerd adventure as there isnt even
updates for the game at all

**Comment by ApacheThunder** [comment]
How did you acquire the game? Downloaded from eShop or from CIA? If from eShop it may be a different story.

**Comment by Shadow322** [comment]
Yea 99% fail error on the game

**Comment by Shadow322** [comment]
3dsiso is where i got it

**Comment by ApacheThunder** [comment]
Yeah that's the problem. You'll need to repack the game. Check the eShop page, it should mention what version the game was released at. 0.0.0 with the first number being "major", second being "minor", and last one being "micro" use the matching commands to set this when using makerom to build the CIA. It's the only fix.

**Comment by urherenow** [comment]
@ApacheThunder you just opened a can of worms trying to explain that. Almost nobody knows how to unpack a .cia, let alone manually pack one. They either download a .cia made by someone or use a tool or script that does everything for them.

**Comment by Zetchzie** [comment]
its doing this for pokemon shuffle and moon chronicles that i genuinely got from eshop.

**Comment by RandomIntermition** [comment]
this issue should be closed, it's nothing to do with rxtools and not a problem that rxtools needs to fix. 

for future REF. this is similar to a problem with wii mods. if version number is lower than what nintendo says is the most recent, it will prompt for update.
if you want to get around this permanently your CIA's should be compiled with a ver number higher than ANY official ver.
(e.g. if nintendo says v1.0.0 + update v1.0.1, set your CIA's to v3.0.0. even if you aren't playing with any updates installed, as far as nin is concerned your version is higher and doesn't require any updates.

WARNING: don't set it too high, leave a bit of room so you can install future update CIA's without having to uninstall everything. (they'll need to be set slightly higher than your last update).
